paasta_tools/paasta_metastatus.py

```diff
@@ -271,106 +263,157 @@ def assert_quorum_size(state):
-                                   naturalsize(resources_total['disk'] * 1024 * 1024, gnu=True))
-                    )
-                else:
-                    formatted_line = (
-                        attribute_location,
-                        '%.2f/%.2f' % (resources_remaining['cpus'], resources_total['cpus']),
-                        '%.2f/%.2f' % (resources_remaining['mem'], resources_total['mem']),
-                        '%.2f/%.2f' % (resources_remaining['disk'], resources_total['disk'])
-                    )
-                rows.append(formatted_line)
-    if len(rows) == 0:
-        return HealthCheckResult(message="  No slave attributes that apply to more than one slave were detected.",
-                                 healthy=True)
-    else:
-        return HealthCheckResult(message='\n'.join(('    %s' % row for row in format_table(rows)))[2:], healthy=True)
 
+def get_resource_utilization_by_attribute(mesos_state, attribute):
+    """
+    Given mesos state and an attribute, calculate resource utilization
+    for each value of a given attribute.
+
+    :param mesost_state: the mesos state
+    :param attribute: the attribute to group slaves by
+    :returns: a dict of {attribute_value: resource_usage}, where resource usage is
+    the dict returned by ``calculate_resource_utilization_for_slaves`` for slaves
+    grouped by attribute value.
+    """
+    slaves = mesos_state.get('slaves', [])
+    if not has_registered_slaves(mesos_state):
+        raise ValueError("There are no slaves registered in the mesos state.")
 
-def slaves_registered(mesos_state):
-    return 'slaves' in mesos_state and mesos_state['slaves']
+    tasks = get_all_tasks_from_state(mesos_state)
+    slave_groupings = group_slaves_by_attribute(slaves, attribute)
+
+    return {
+        attribute_value: calculate_resource_utilization_for_slaves(slaves, tasks)
+        for attribute_value, slaves in slave_groupings
+    }
 
 
-def get_mesos_status(mesos_state, verbosity, humanize_output=False):
-    """Gathers information about the mesos cluster.
-       :return: tuple of a string containing the status and a bool representing if it is ok or not
+def resource_utillizations_from_resource_info(total, free):
+    """
+    Given two ResourceInfo tuples, one for total and one for free,
+    create a ResourceUtilization tuple for each metric in the ResourceInfo.
+    :param total:
+    :param free:
+    :returns: ResourceInfo for a metric
     """
+    return [
+        ResourceUtilization(metric=field, total=total[index], free=free[index])
+        for index, field in enumerate(ResourceInfo._fields)
+    ]
 
-    cluster_results = run_healthchecks_with_param(mesos_state, [assert_quorum_size, assert_no_duplicate_frameworks])
 
-    metrics = get_mesos_stats()
-    metrics_results = run_healthchecks_with_param(metrics, [
+def has_registered_slaves(mesos_state):
+    return 'slaves' in mesos_state and mesos_state['slaves']
```

Comment:
This reads odd to me. Just a bikeshed, but how about

    len(mesos_state.get('slaves', []) > 0